packages/flutter_tools/lib/src/commands/attach.dart

```diff
@@ -218,11 +218,17 @@ known, it can be explicitly provided to attach via the command-line, e.g.
     MacOSDesignedForIPadDevices.allowDiscovery = true;
 
     await super.validateCommand();
-    if (await findTargetDevice() == null) {
+
+    final Device? targetDevice = await findTargetDevice();
+    if (targetDevice == null) {
       throwToolExit(null);
     }
+
     debugPort;
-    if (debugPort == null && debugUri == null && argResults!.wasParsed(FlutterCommand.ipv6Flag)) {
+    if (targetDevice is FuchsiaDevice &&
```

Comment:
Seems like the help text needs to be updated then?
https://github.com/flutter/flutter/blob/9d820aa35a22ef0365d3814462dc11f56afc13c9/packages/flutter_tools/lib/src/runner/flutter_command.dart#L606-L607

Or should this be `targetDevice is! IOSDevice`?  It looks like on Android it does do the log scanning thing to determine it.

Reply:
Where did you see it get determined for Android?

Reply:
It looks like this codepath is used for Android, it gets the port from log scanning:
https://github.com/flutter/flutter/blob/f9857df6b3168bd93b68806eeff9313eff373b78/packages/flutter_tools/lib/src/protocol_discovery.dart#L90-L96
and then the uri logic uses `ipv6` to swap out the host if needed:
https://github.com/flutter/flutter/blob/f9857df6b3168bd93b68806eeff9313eff373b78/packages/flutter_tools/lib/src/protocol_discovery.dart#L147-L149